File: mRemoteV1/Forms/OptionsPages/AppearancePage.vb
Imports mRemote3G.App

Namespace Forms.OptionsPages
    Public Class AppearancePage
        Public Overrides Property PageName As String
            Get
                Return Language.Language.strTabAppearance
            End Get
            Set
            End Set
        End Property

        Public Overrides Sub ApplyLanguage()
            MyBase.ApplyLanguage()

            lblLanguage.Text = Language.Language.strLanguage
            lblLanguageRestartRequired.Text = String.Format(Language.Language.strLanguageRestartRequired,
                                                            Application.ProductName)
            chkShowDescriptionTooltipsInTree.Text = Language.Language.strShowDescriptionTooltips
            chkShowFullConnectionsFilePathInTitle.Text = Language.Language.strShowFullConsFilePath
            chkShowSystemTrayIcon.Text = Language.Language.strAlwaysShowSysTrayIcon
            chkMinimizeToSystemTray.Text = Language.Language.strMinimizeToSysTray
        End Sub

        Public Overrides Sub LoadSettings()
            MyBase.SaveSettings()

            cboLanguage.Items.Clear()
            cboLanguage.Items.Add(Language.Language.strLanguageDefault)

            For Each nativeName As String In SupportedCultures.CultureNativeNames
                cboLanguage.Items.Add(nativeName)
            Next
            If _
                Not String.IsNullOrEmpty(My.Settings.OverrideUICulture) AndAlso
                SupportedCultures.IsNameSupported(My.Settings.OverrideUICulture) Then
                cboLanguage.SelectedItem = SupportedCultures.CultureNativeName(My.Settings.OverrideUICulture)
            End If
            If cboLanguage.SelectedIndex = - 1 Then
                cboLanguage.SelectedIndex = 0
            End If

            chkShowDescriptionTooltipsInTree.Checked = My.Settings.ShowDescriptionTooltipsInTree
            chkShowFullConnectionsFilePathInTitle.Checked = My.Settings.ShowCompleteConsPathInTitle
            chkShowSystemTrayIcon.Checked = My.Settings.ShowSystemTrayIcon
            chkMinimizeToSystemTray.Checked = My.Settings.MinimizeToTray
        End Sub

        Public Overrides Sub SaveSettings()
            MyBase.SaveSettings()

            If cboLanguage.SelectedIndex > 0 And SupportedCultures.IsNativeNameSupported(cboLanguage.SelectedItem) Then
                My.Settings.OverrideUICulture = SupportedCultures.CultureName(cboLanguage.SelectedItem)
            Else
                My.Settings.OverrideUICulture = String.Empty
            End If

            My.Settings.ShowDescriptionTooltipsInTree = chkShowDescriptionTooltipsInTree.Checked
            My.Settings.ShowCompleteConsPathInTitle = chkShowFullConnectionsFilePathInTitle.Checked
            frmMain.ShowFullPathInTitle = chkShowFullConnectionsFilePathInTitle.Checked

            My.Settings.ShowSystemTrayIcon = chkShowSystemTrayIcon.Checked
            If My.Settings.ShowSystemTrayIcon Then
                If Runtime.NotificationAreaIcon Is Nothing Then
                    Runtime.NotificationAreaIcon = New Tools.Controls.NotificationAreaIcon
                End If
            Else
                If Runtime.NotificationAreaIcon IsNot Nothing Then
                    Runtime.NotificationAreaIcon.Dispose()
                    Runtime.NotificationAreaIcon = Nothing
                End If
            End If

            My.Settings.MinimizeToTray = chkMinimizeToSystemTray.Checked
        End Sub
    End Class
End Namespace

File: mRemoteV1/CS/Forms/OptionsPages/AppearancePage.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using mRemoteNG.App;
using mRemoteNG.My;


namespace mRemoteNG.Forms.OptionsPages
{
	public partial class AppearancePage
		{
			public AppearancePage()
			{
				InitializeComponent();
			}
public override string PageName
			{
				get
				{
					return Language.strTabAppearance;
				}
				set
				{
				}
			}
			
			public override void ApplyLanguage()
			{
				base.ApplyLanguage();
				
				lblLanguage.Text = Language.strLanguage;
				lblLanguageRestartRequired.Text = string.Format(Language.strLanguageRestartRequired, Application.Info.ProductName);
				chkShowDescriptionTooltipsInTree.Text = Language.strShowDescriptionTooltips;
				chkShowFullConnectionsFilePathInTitle.Text = Language.strShowFullConsFilePath;
				chkShowSystemTrayIcon.Text = Language.strAlwaysShowSysTrayIcon;
				chkMinimizeToSystemTray.Text = Language.strMinimizeToSysTray;
			}
			
			public override void LoadSettings()
			{
				base.SaveSettings();
				
				cboLanguage.Items.Clear();
				cboLanguage.Items.Add(Language.strLanguageDefault);
				
				foreach (string nativeName in SupportedCultures.CultureNativeNames)
				{
					cboLanguage.Items.Add(nativeName);
				}
				if (!string.IsNullOrEmpty(Settings.OverrideUICulture) && SupportedCultures.IsNameSupported(Settings.OverrideUICulture))
				{
					cboLanguage.SelectedItem = SupportedCultures.get_CultureNativeName(Settings.OverrideUICulture);
				}
				if (cboLanguage.SelectedIndex == -1)
				{
					cboLanguage.SelectedIndex = 0;
				}
				
				chkShowDescriptionTooltipsInTree.Checked = Settings.ShowDescriptionTooltipsInTree;
				chkShowFullConnectionsFilePathInTitle.Checked = Settings.ShowCompleteConsPathInTitle;
				chkShowSystemTrayIcon.Checked = Settings.ShowSystemTrayIcon;
				chkMinimizeToSystemTray.Checked = Settings.MinimizeToTray;
			}
			
			public override void SaveSettings()
			{
				base.SaveSettings();
				
				if (cboLanguage.SelectedIndex > 0 && SupportedCultures.IsNativeNameSupported(System.Convert.ToString(cboLanguage.SelectedItem)))
				{
					Settings.OverrideUICulture = SupportedCultures.get_CultureName(System.Convert.ToString(cboLanguage.SelectedItem));
				}
				else
				{
					Settings.OverrideUICulture = string.Empty;
				}
				
				Settings.ShowDescriptionTooltipsInTree = chkShowDescriptionTooltipsInTree.Checked;
				Settings.ShowCompleteConsPathInTitle = chkShowFullConnectionsFilePathInTitle.Checked;
				frmMain.Default.ShowFullPathInTitle = chkShowFullConnectionsFilePathInTitle.Checked;
				
				Settings.ShowSystemTrayIcon = chkShowSystemTrayIcon.Checked;
				if (Settings.ShowSystemTrayIcon)
				{
					if (Runtime.NotificationAreaIcon == null)
					{
						Runtime.NotificationAreaIcon = new Tools.Controls.NotificationAreaIcon();
					}
				}
				else
				{
					if (Runtime.NotificationAreaIcon != null)
					{
						Runtime.NotificationAreaIcon.Dispose();
						Runtime.NotificationAreaIcon = null;
					}
				}
				
				Settings.MinimizeToTray = chkMinimizeToSystemTray.Checked;
			}
		}
}
